pkg/asset/quota/aws/instancetypes.go

```diff
@@ -18,39 +20,39 @@ type InstanceTypeInfo struct {
 
 // InstanceTypes returns information of the all the instance types available for a region.
 // It returns a map of instance type name to it's information.
-func InstanceTypes(ctx context.Context, sess *session.Session, region string) (map[string]InstanceTypeInfo, error) {
+func InstanceTypes(ctx context.Context, region string, serviceEndpoints []typesaws.ServiceEndpoint) (map[string]InstanceTypeInfo, error) {
 	ret := map[string]InstanceTypeInfo{}
 
-	client := ec2.New(sess, aws.NewConfig().WithRegion(region))
```

Comment:
I think this is what you're referring to with:

> Questions: User-provided service endpoints were never considered here. Is this intentional?

No, I don't think this is intentional. It is a good example of one of the merits of 
[CORS-3793](https://issues.redhat.com//browse/CORS-3793). If we encapsulate client creation, we don't need to remember to provide service endpoints at each calling site.